Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -80,36 +103,30 @@ def _create_engine_and_connect(self) -> sqlalchemy.engine.base.Connection:
         Creating and engine according to the instance preferences and connecting
         :return: a connection object that will be used in order to execute SQL queries
         """
-        module = self._convert_dialect_to_module(self.dialect)
-        port_part = ''
-        encoded_password = urllib.parse.quote_plus(self.password)
-        if self.port:
-            port_part = f':{self.port}'
-        db_preferences = f'{module}://{self.username}:{encoded_password}@{self.host}{port_part}/{self.dbname}'
         ssl_connection = {}
-        if self.dialect == "Microsoft SQL Server":
-            db_preferences += "?driver=FreeTDS"
-        if self.connect_parameters and self.dialect == "Microsoft SQL Server":
-            db_preferences += f'&{self.connect_parameters}'
-        elif self.connect_parameters and self.dialect != "Microsoft SQL Server":
-            # a "?" was already added when the driver was defined
-            db_preferences += f'?{self.connect_parameters}'
-
+        module = self._convert_dialect_to_module(self.dialect)
+        db_url = URL(drivername=module,
+                     username=self.username,
+                     password=self.password,
+                     host=self.host,
+                     port=self.port,
```

Comment:
Need to make sure we don't have an empty string or 0
```suggestion
                     port=self.port or None,
```

Reply:
This case is handled in the `generate_default_port_by_dialect` method.

In such case the `not port` bool will be satisfied and the method will be executed, and it will return `None`

Reply:
ok